bundle

```diff
@@ -128,29 +140,33 @@ def version_bundle(bundle, channel, overrides):
                 copy['applications'][application]['charm'] = name + '-' + revision
                 break
         else:
-            resp = requests.get('https://api.jujucharms.com/charmstore/v5/meta/id',
-                                params={'id': charm, 'channel': channel})
-            if not resp.ok:
-                err = "Couldn't query %s %s, is it public?" % (charm, channel)
-                raise CharmstoreQueryError(err)
-            meta = resp.json()
-            meta = [meta[k] for k in meta][0]
-            copy['applications'][application]['charm'] = meta['Id']
+            if charm.startswith("cs:"):
+                resp = requests.get('https://api.jujucharms.com/charmstore/v5/meta/id',
+                                    params={'id': charm, 'channel': channel})
+                if not resp.ok:
+                    err = "Couldn't query %s %s, is it public?" % (charm, channel)
+                    raise CharmstoreQueryError(err)
+                meta = resp.json()
+                meta = [meta[k] for k in meta][0]
+                copy['applications'][application]['charm'] = meta['Id']
+            else:
+                copy['applications'][application]['channel'] = channel
 
     # pin resource revisions
     for application in bundle['applications']:
         charm = bundle['applications'][application]['charm']
-        resp = requests.get('https://api.jujucharms.com/charmstore/v5/meta/resources',
-                            params={'id': charm, 'channel': channel})
-        if not resp.ok:
-            err = "Couldn't query resources for %s %s, is it public?" % (charm, channel)
-            raise CharmstoreQueryError(err)
-        response_data = resp.json()
-        copy['applications'][application].setdefault('resources', {})
-        for resource in response_data[charm]:
-            name = resource['Name']
-            revision = resource['Revision']
-            copy['applications'][application]['resources'][name] = revision
+        if charm.startswith("cs:"):
+            resp = requests.get('https://api.jujucharms.com/charmstore/v5/meta/resources',
+                                params={'id': charm, 'channel': channel})
+            if not resp.ok:
+                err = "Couldn't query resources for %s %s, is it public?" % (charm, channel)
+                raise CharmstoreQueryError(err)
+            response_data = resp.json()
+            copy['applications'][application].setdefault('resources', {})
+            for resource in response_data[charm]:
+                name = resource['Name']
+                revision = resource['Revision']
+                copy['applications'][application]['resources'][name] = revision
```

Comment:
charmhub charms have revisions pinned per track -- therefore it's not necessary to specify those revisions in the bundle